scipy/linalg/_basic.py

```diff
@@ -146,6 +152,33 @@ def solve(a, b, lower=False, overwrite_a=False,
     array([ True,  True,  True], dtype=bool)
 
     """
+    xp = array_namespace(a, b)
+    if is_numpy(xp):
+        return _solve(a, b, lower=lower, overwrite_a=overwrite_a,
+                      overwrite_b=overwrite_b, check_finite=check_finite,
+                      assume_a=assume_a, transposed=transposed)
+    if lower:
+        raise ValueError(arg_err_msg("lower"))
+    if overwrite_a:
```

Comment:
should I be using `elif` for these parameter checks instead of `if`? Clearly they are logically equivalent here, but maybe `elif` has better performance, or would be more readable?